dotnet/samples/Concepts/Planners/AutoFunctionCallingPlanning.cs

```diff
@@ -0,0 +1,350 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Concurrent;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.Json;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+using Microsoft.SemanticKernel.Planning;
+
+namespace Planners;
+
+/// <summary>
+/// This example shows how to implement plan generation and execution with Auto Function Calling and
+/// enable telemetry, filters and caching.
+/// <see cref="ChatHistory"/> object is used for plan manipulation and execution.
+/// </summary>
+public class AutoFunctionCallingPlanning(ITestOutputHelper output) : BaseTest(output)
```

Comment:
Remember to update the concepts README to point at the new file.

Reply:
Thanks! It would be nice to automate this process at some point 😄